packages/cells/src/widget.ts

```diff
@@ -1445,8 +1457,8 @@ export class MarkdownCell extends AttachmentsCell {
   /**
    * Modify the cell source to include a reference to the attachment.
    */
-  protected updateCellSourceWithAttachment(attachmentName: string) {
-    const textToBeAppended = `![${attachmentName}](attachment:${attachmentName})`;
+  protected updateCellSourceWithAttachment(attachmentName: string, URI?: string) {
+    const textToBeAppended = `![${attachmentName}](attachment:${URI ?? attachmentName})`;
```

Comment:
To be safe, I suppose this ought to be
```suggestion
    const textToBeAppended = `![${attachmentName}](attachment:${URI ?? encodeURI(attachmentName)})`;
```